### PR TITLE
Remove compiler version cache

### PR DESCRIPTION
**Pull request by rmoyard**
**Description of the Change:**
- Remove unnecessary fixture
- Remove the compiler cache

### CODE REVIEW

File: pennylane/compiler/compiler.py [hunk 1]
```diff
@@ -66,12 +66,12 @@ def _check_compiler_version(name):
 
     if name == "catalyst":
         installed_catalyst_version = metadata.version("pennylane-catalyst")
+        name = name + installed_catalyst_version
```
Comment (dime10) [inline pennylane/compiler/compiler.py:69]: Doesn't this break future checks in this function?

Comment (rmoyard) [reply pennylane/compiler/compiler.py:69]: Yes goo catch I am changing that

Comment (dime10) [reply pennylane/compiler/compiler.py]: What is the motivation behind the change?

**Review by timmysilv — APPROVED** [review]
love a good `+0` PR. I hope that `metadata.version` doesn't incur a notable performance hit, but we can always add this back carefully if it does. for now, let's unblock master!

File: pennylane/compiler/compiler.py (hunk 1 above)
Comment (dime10) [reply pennylane/compiler/compiler.py]: I've had a look, we seem to only invoke the check in calls to `qml.qjit`, so any potential overhead due to the removal of the cache is likely irrelevant 👍 

Double checking with @maliasadi that it's ok to remove

**Review by maliasadi — APPROVED** [review]
Thanks @rmoyard for the quick fix!

We added the `compiler_checked` map to store the backend compilers versions and reduce extra calls to `metadata.version` and `semantic_version.Version`. Without re-using the cached version, every call to `qml.qjit` will have an overhead of ~`0.0003` second (running on a 13th Gen Intel i7). This can be considered negligible if users may try downgrading the Catalyst version in some scenarios.

However, regarding the current failed tests, they are related to mokey-patching the Catalyst version in the following test:
https://github.com/PennyLaneAI/pennylane/blob/badb494e3e74d5c99b506b0ff5ab8a7612889220/tests/test_compiler.py#L37
Removing this test should fix the issue.

We chatted offline and convinced that users may downgrade the Catalyst version and this issue is more bothering that the none-caching overhead! So, let's merge this patch as a quick fix to unblock PL and we can revisit this later when we have a better solution for it.

**Pull request merged**